lib/watir/rails/middleware.rb

```diff
@@ -2,25 +2,49 @@ module Watir
   class Rails
     # @private
     class Middleware
+      class Counter
+        attr_reader :value
+
+        def initialize
+          @value = 0
+          @mutex = Mutex.new
+        end
+
+        def increment
+          @mutex.synchronize { @value += 1 }
+        end
+
+        def decrement
+          @mutex.synchronize { @value -= 1 }
+        end
+      end
+
       attr_accessor :error
 
       def initialize(app)
         @app = app
+        @counter = Counter.new
+      end
+
+      def pending_requests?
+        @counter.value.positive?
```

Comment:
Isn't ` > 0` more intuitive? :P I had to go back and check what type `value` was since I expected it to be a number and it turns out it was a number, but I had to double check just in case I missed something above.

Reply:
Somehow RuboCop suggests to use `#positive?`, but I'll use `> 0`.